src/core/include/openvino/core/except.hpp

```diff
@@ -145,7 +145,7 @@ class OPENVINO_API NotImplemented : public AssertFailure {
 //
 #define OPENVINO_ASSERT_HELPER2(exc_class, ctx, check, ...)                      \
     do {                                                                         \
-        if (!(check)) {                                                          \
+        if (!static_cast<bool>(check)) {                                         \
```

Comment:
@praasz Could you please take a look.
I think this is in general a good change even outside of the context of the current one.

Reply:
See also [comment](https://github.com/openvinotoolkit/openvino/pull/31342#discussion_r2239093923).